.github/workflows/prerelease_windows.yml

```diff
@@ -76,3 +81,39 @@ jobs:
       - name: Upload MSI to GH
         shell: bash
         run: build/upload_artifacts_gh.sh
+
+  publishing-to-s3:
+    name: Publish windows artifacts into s3 test bucket
+    runs-on: ubuntu-20.04
+    needs: [packaging-windows]
+
+    steps:
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ env.DOCKER_HUB_ID }}
+          password: ${{ env.DOCKER_HUB_PASSWORD }}
+      - name: Publish to S3 action
+        uses: newrelic/infrastructure-publish-action@main
+        env:
+          AWS_S3_BUCKET_NAME: ${{ secrets.TEST_AWS_S3_BUCKET_NAME }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.TEST_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.TEST_AWS_SECRET_ACCESS_KEY }}
+        with:
+          tag: ${{env.TAG}}
+          app_name: "newrelic-infra"
+          repo_name: "newrelic/infrastructure-agent"
+          env: pre-release
+          # 'custom' is for custom schema file
+          schema: "custom"
+          schema_url: "https://raw.githubusercontent.com/newrelic/infrastructure-agent/ci/pre-release-publish/build/upload-schema-windows.yml"
+          # careful with the access_keys used as it can inadvertently publish to where we don't want,
+          # especially if the bucket name is "common" enough that it can already be used for something else
+          # note: seems like using secrets here directly seems to fail for some cases
```

Comment:
we could rewrite this note to, credentials for the action need to be a  AWS STS key pair ready to use assume IAM role  https://github.com/newrelic/infrastructure-publish-action/blob/main/action.yml#L16-L18

see STS handshake at https://github.com/newrelic/infrastructure-publish-action/blob/main/scripts/Makefile#L67-L71

Reply:
BTW we are missing `aws_role_arn` here